nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/state/flow/flow.effects.ts

```diff
@@ -2056,4 +2058,277 @@ export class FlowEffects {
+                        ]).pipe(
+                            map(([stopPgResponse]) => {
+                                return FlowActions.stopComponentSuccess({
+                                    response: {
+                                        type: request.type,
+                                        component: stopPgResponse
+                                    }
+                                });
+                            })
+                        );
+                    default:
+                        return of(FlowActions.flowApiError({ error: `${request.type} does not support stopping` }));
+                }
+            }),
+            catchError((error) => of(FlowActions.flowApiError({ error: error.error })))
+        )
+    );
+
+    /**
+     * If the component stopped was the current process group, reload the flow
+     */
+    stopCurrentProcessGroupSuccess$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.stopComponentSuccess),
+            map((action) => action.response),
+            withLatestFrom(this.store.select(selectCurrentProcessGroupId)),
+            filter(([response, currentPg]) => response.component.id === currentPg),
+            switchMap(() => of(FlowActions.reloadFlow()))
+        )
+    );
+
+    /**
+     * If a ProcessGroup was stopped, it should be reloaded as the response from the stop operation doesn't contain all the displayed info
+     */
+    stopProcessGroupSuccess$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.stopComponentSuccess),
+            map((action) => action.response),
+            withLatestFrom(this.store.select(selectCurrentProcessGroupId)),
+            filter(([response, currentPg]) => response.component.id !== currentPg),
+            filter(([response]) => response.type === ComponentType.ProcessGroup),
+            switchMap(([response]) =>
+                of(
+                    FlowActions.reloadProcessGroup({
+                        request: {
+                            id: response.component.id
+                        }
+                    })
+                )
+            )
+        )
+    );
+
+    runOnce$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.runOnce),
+            map((action) => action.request),
+            switchMap((request) => {
+                return from(this.flowService.runOnce(request)).pipe(
+                    map((response) =>
+                        FlowActions.runOnceSuccess({
+                            response: {
+                                component: response
+                            }
+                        })
+                    )
+                );
+            }),
+            catchError((error) => of(FlowActions.flowApiError({ error: error.error })))
+        )
+    );
+
+    runOnceSuccess$ = createEffect(() =>
```

Comment:
I think we can just get rid of this entire effect since we were able to update the store based on the response to the run once API call.